pkg/plugins/runtime/k8s/controllers/gateway_converter.go

```diff
@@ -0,0 +1,114 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	kube_core "k8s.io/api/core/v1"
+	kube_apierrs "k8s.io/apimachinery/pkg/api/errors"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	kube_controllerutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	mesh_k8s "github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/metadata"
+	k8s_util "github.com/kumahq/kuma/pkg/plugins/runtime/k8s/util"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+// createorUpdateBuiltinGatewayDataplane manages the dataplane for a pod
+// belonging to a built-in Kuma gateway.
+func (r *PodReconciler) createorUpdateBuiltinGatewayDataplane(ctx context.Context, pod *kube_core.Pod) error {
+	dataplane := &mesh_k8s.Dataplane{
+		ObjectMeta: kube_meta.ObjectMeta{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+		},
+		Mesh: k8s_util.MeshFor(pod),
+	}
+
+	builtinName, exists := metadata.Annotations(pod.Annotations).GetString(metadata.KumaBuiltinGatewayAnnotation)
+	if !exists || builtinName == "" {
+		return nil
+	}
+
+	dataplaneProto, err := r.PodConverter.BuiltinGatewayDataplane(pod, builtinName)
+	if err != nil {
+		return errors.Wrap(err, "unable to translate a Gateway Pod into a Dataplane")
+	} else if dataplaneProto == nil {
+		// we don't want a dataplane, the existing object will be deleted
+		// through owner refs
+		return nil
+	}
+
+	spec, err := util_proto.ToMap(dataplaneProto)
+	if err != nil {
+		return err
+	}
+
+	operationResult, err := kube_controllerutil.CreateOrUpdate(ctx, r.Client, dataplane, func() error {
+		dataplane.Spec = spec
+
+		if err := kube_controllerutil.SetControllerReference(pod, dataplane, r.Scheme); err != nil {
+			return errors.Wrap(err, "unable to set Dataplane's controller reference to Pod")
```

Comment:
Probably should ignore and `AlreadyOwned` error here?

Reply:
This is now in `master` but shouldn't we not ignore an `AlreadyOwned` since the owner should refer to us?